crates/red_knot_python_semantic/src/types.rs

```diff
@@ -1108,7 +1108,6 @@ impl<'db> Type<'db> {
                             possibly_unbound = true;
                         }
                         Symbol::Type(ty_member, member_boundness) => {
-                            // TODO: raise a diagnostic if member_boundness indicates potential unboundness
```

Comment:
As the tests show, we *do* raise a diagnostic if `member_boundness` indicates potential unboundness:
```py
class C1:
    x = 1

class C2:
    if bool_instance():
        x = 2

class C3:
    x = 3

flag1 = bool_instance()
flag2 = bool_instance()

C = C1 if flag1 else C2 if flag2 else C3

# error: [possibly-unbound-attribute] "Attribute `x` on type `Literal[C1, C2, C3]` is possibly unbound"
reveal_type(C.x)  # revealed: Literal[1, 2, 3]
```

So this comment can be removed.